src/app/index.js

```diff
@@ -10,38 +10,83 @@ angular.module('app', [
-  '$locationProvider',
-  ($stateProvider, $urlRouterProvider, $locationProvider) => {
-    // Allow the use of HTML5 state modes
-    $locationProvider.html5Mode(true)
-
-    // Define all of the application states
-    $stateProvider
-      .state('home', {
-        url: '/',
-        template: require('./components/home/homeView.html'),
-        controller: 'HomeController',
-        controllerAs: 'vm'
-      })
-      .state('login', {
-        url: '/login',
-        template: require('./components/login/loginView.html'),
-        controller: 'LoginController',
-        controllerAs: 'vm'
-      })
-      .state('event', {
-        url: '/event',
-        template: require('./components/event/eventView.html'),
-        controller: 'EventController',
-        controllerAs: 'vm'
-      })
-
-    // If no states were matched, redirect to home
-    $urlRouterProvider.otherwise('home')
-  }])
+]).constant('_', _)
+  .config([
+    '$stateProvider',
+    '$urlRouterProvider',
+    '$locationProvider',
+    '$authProvider',
+    '$windowProvider',
+    ($stateProvider, $urlRouterProvider, $locationProvider, $authProvider, $windowProvider) => {
+      // Allow the use of HTML5 state modes
+      $locationProvider.html5Mode(true)
+
+      let loginRequired = ['$q', '$location', '$auth', ($q, $location, $auth) => {
+        let deferred = $q.defer()
+        if ($auth.isAuthenticated()) {
+          deferred.resolve()
+        } else {
+          $location.path('/login')
+        }
+        return deferred.promise
+      }]
+
+      // Define all of the application states
+      $stateProvider
+        .state('home', {
+          url: '/',
+          template: require('./components/home/homeView.html'),
+          controller: 'HomeController',
+          controllerAs: 'homeCtrl',
+          onEnter: ['$state', '$auth', ($state, $auth) => {
+            if ($auth.isAuthenticated()) {
+              $state.go('dashboard')
+            }
+          }]
+        })
+        .state('dashboard', {
+          url: '/dashboard',
+          template: require('./components/dashboard/dashboardView.html'),
+          controller: 'DashboardController',
+          controllerAs: 'dashCtrl',
+          resolve: {
+            loginRequired: loginRequired
+          }
+        })
+        .state('login', {
+          url: '/login',
+          template: require('./components/login/loginView.html'),
+          controller: 'LoginController',
+          controllerAs: 'loginCtrl',
+          onEnter: ['$state', '$auth', ($state, $auth) => {
+            if ($auth.isAuthenticated()) {
+              $state.go('dashboard')
+            }
+          }]
+        })
+        .state('logout', {
+          url: '/logout',
+          onEnter: ['$state', '$auth', ($state, $auth) => {
+            $auth.logout()
+            $state.go('login')
+          }]
+        })
+
+      // If no states were matched, redirect to home
+      $urlRouterProvider.otherwise('enter')
```

Comment:
Good catch! I experimented with adding an 'enter' state, and forgot to revert that condition when I decided against it. Fixed.